design/user_data_incentives.md

```diff
@@ -0,0 +1,124 @@
+
+The Obscuro protocol is designed in such a way that it decouples the income from the costs by maintaing a buffer.
+
+We can use this designed mechanism to pay for node usage as well along with the L1 gas fees and the general incentives to follow the protocol.
+
+
+##### Measuring node usage
+
+As a proxy for a node responding to user requests, we can use a model where a node is payed a percentage of the gas fees that originated from their node. 
+
+A user that is connected to a node that doesn't respond to requests (like transaction receipts, or events), will leave that node and connect to a different node.
+
+The "Wallet extension", our client-side software can hide this complexity from the user, and switch nodes without the user noticing.
+
+This is not a perfect system, but it should be a good approximation. It turns a node operator into a service provider that makes an income based on the quality of service it provides. 
+
+It also removes the need to incentivise verifiers to break the "verifier's dilemma"
+
+
+##### Technical implementation
+
+To implement the mechanism described above, the network needs to keep track of the original node that received the transaction via RPC.
+
+Upon receiving a signed transaction from the wallet (MetaMask), the wallet extension will create a payload like this:
+
+```
+type RPCTxReceiver struct {
+  SignedTransaction []byte   // The signed transaction generated by the wallet.
+  RPCNode           Address  // The address of the node that accepted the request.
+  R, S              *big.Int // Signature by the registered viewing key of the transaction signer.
+}
+```
+
+The transaction is wrapped together with a node address, and signed with the viewing key. 
+
+A node will only accept transactions via RPC if they're addressed to it, and thus can use this payload to claim a reward. 
+This wrapper will then be gossiped to the other nodes accompanied by the signed viewing key.
+
+
+##### Outstanding problems to solve
+
+Problem 1: a malicious or malfunctioning wallet extension could submit the same transction wrapped for multiple nodes.
+This is not a major problem because the incentives to do this are not large
+
+Problem 2: how to include the transactions in the rollup. 
+It is not ideal to include this extra wrapped information in the rollup and bulk it up. 
+
+If the extra info is not included then it becomes harder to verify whether the fees were allcated correctly.
+Nodes could also claim in bulk by constructing a ZKP that is included.
+
+
+##### Payout
+
+Assuming we solve the claiming problem, the question is how much should a node receive per transaction.
+
+- A fixed amount per transaction
+- A percentage of the fee paid by the transaction 
+
+What amount would make running an L2 node that offers a useful service a reasonably profitable business?
+Is that amount sustainable from the fees collected by the network? Is thre any help required from the foundation?
+
```

Comment:
This needs modelling ;-)
My initial reaction is a fixed amount would be difficult to get right whereas a percentage could be easier to manage from an economics POV. But it might be more difficult to implement technically?

Reply:
Definitely needs modelling. Let's see what makes the most sense, and then we'll assess how difficult it is